pkg/apiclient/logs-intermediary.go

```diff
@@ -25,6 +27,13 @@ func (c *logsIntermediary) Recv() (*workflowpkg.LogEntry, error) {
 	}
 }
 
+func (c *logsIntermediary) SendHeader(metadata.MD) error {
+	// We invoke `SendHeader` in order to eagerly flush headers to allow us to send period
+	// keepalives when using HTTP/1 and Server Sent Events, so we need to implement this here,
+	// though we don't use the meta for anything.
+	return nil
```

Comment:
Like the other location, I'd appreciate extra scrutiny here since I'm not familiar enough with gRPC to know what implications this might have

Reply:
I don't think the intermediary should need do this - they're just glue. If it is needed, I think it probably is needed else where

Reply:
This was added because the default implementation of the intermediary just panics - implementing this fixed the test failures since if I understand what they're doing correctly, the intermediaries are passed as the`ws` arg in 

https://github.com/argoproj/argo-workflows/blob/b60ab0387eb2895ccb161890c32213a3538c898d/server/workflow/workflow_server.go#L180

but I'm not going to pretend to know 100% if there is something else I should do here.  We invoke `SendHeader` in the streaming functions because otherwise the Stream Forwarders are not invoked until the first time we receive data which means keepalives won't work.

If there's an alternate suggestion to workaround that, I'm open to suggestions since I don't know much about gRPC at the moment.